scripts/sniffer-testsuite.test

```diff
@@ -77,16 +83,42 @@ fi
 if test $RESULT -eq 0 && test $has_rsa == yes && test $has_tlsv12 == yes && test $has_static_rsa == yes
 then
     echo -e "\nStaring snifftest on sniffer-ipv6.pcap...\n"
-    ./sslSniffer/sslSnifferTest/snifftest ./scripts/sniffer-ipv6.pcap ./certs/server-key.pem ::1 11111
+    ./sslSniffer/sslSnifferTest/snifftest -pcap ./scripts/sniffer-ipv6.pcap -key ./certs/server-key.pem -server ::1 -port 11111
 
     RESULT=$?
     [ $RESULT -ne 0 ] && echo -e "\nsnifftest (ipv6) failed\n" && exit 1
 fi
 
+#  TLS v1.2 sniffer keylog file test: runs sniffer on pcap and associated keylog file and compares decrypted traffic with known good output.
+#  To regenerate the known good output, run `scripts/sniffer-gen.sh` to regenerate the pcap and keylog file, then run the sniffer on it
+#  with the same arguments as in the test belowl, but redirect output to `./scripts/sniffer-tls12-keylog.out`.
+if test $RESULT -eq 0 && test $has_tlsv13 == yes && test $has_keylog == yes
+then
+    TMPFILE=$(mktemp)
+    RESULT=$?
+    [ $RESULT -ne 0 ] && echo -e "\nsnifftest keylog test failed: unable to create tmpfile\n" && rm $TMPFILE && exit 1
+
+    ./sslSniffer/sslSnifferTest/snifftest \
```

Comment:
Is this expected output?

```
% ./sslSniffer/sslSnifferTest/snifftest \
         -pcap scripts/sniffer-tls12-keylog.pcap \
         -keylogfile scripts/sniffer-tls12-keylog.sslkeylog \
         -server 127.0.0.1 -port 11111
snifftest 5.6.3
sniffer features: key_callback tls_v13 tls_v12 static_ephemeral sni extended_master rsa dh ecc rsa_static dh_static ssl_keylog_file

Found duplicate client random value in keylog file. Rejecting
Found duplicate client random value in keylog file. Rejecting
Found duplicate client random value in keylog file. Rejecting
Found duplicate client random value in keylog file. Rejecting
Found duplicate client random value in keylog file. Rejecting
Found duplicate client random value in keylog file. Rejecting
SSL App Data(26:14):hello wolfssl!
SSL App Data(27:22):I hear you fa shizzle!
SSL App Data(57:14):hello wolfssl!
SSL App Data(58:22):I hear you fa shizzle!
```

Reply:
Yes, the keylog file has duplicates because the pcap/keylog file were generated from a client/server session where both client and server were built with `--enable-keylog-export (`SHOW_SECRETS`, `HAVE_SECRET_CALLBACK`, etc) and so both ends dump their secrets to the keylog file when handshake happens. This is totally fine and the sniffer can handle this, it just warns that it sees more than one entry in the log file for a given client random.